ui/src/components/References/GroupReference.tsx

```diff
@@ -23,6 +26,19 @@ export default function GroupReference({ flag }: GroupReferenceProps) {
     }
   }, [gang, group, flag]);
 
+  const buttonMessage = () => {
+    if (group) {
+      return 'Go';
+    }
+    if (requested) {
+      return 'Join Requested';
+    }
+    if (privacy === 'private') {
+      return 'Request to Join';
+    }
+    return 'Join';
+  };
+
```

Comment:
we already have this logic if you just use `button.text` from `useGroupJoin`, maybe we should it's language out for these though

Reply:
Thanks, missed that in useGroupJoin.